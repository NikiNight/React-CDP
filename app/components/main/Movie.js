import React from 'react';

class Movie extends React.Component {
    render() {
        return (
            <article className="movie">
                <img src="https://via.placeholder.com/300x400.jpg?text=Movie%20Cover" alt="" className="movie__image" />
                <div className="movie__description">
                    <div>
                        <h2 className="movie__title">Jackie Brown</h2>
                        <p className="movie__category">Drama</p>
                    </div>
                    <p className="movie__year">1994</p>
                </div>
            </article>
        )
    }
}

export default Movie;