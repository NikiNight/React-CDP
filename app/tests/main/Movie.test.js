import React from 'react';
import Movie from './../../components/main/Movie';
import renderer from 'react-test-renderer';
import {mount} from 'enzyme';

describe('Test Movie component', () => {

    test('Movie matches snapshot', () => {
        const tree = renderer.create(<Movie />)
          .toJSON();
        expect(tree).toMatchSnapshot();
    });

    it('Should be call selectMovie', () => {
        const selectMovie = jest.fn();
        const component = mount(<Movie selectMovie={selectMovie}/>);
        component.simulate('click');
        expect(selectMovie).toHaveBeenCalled();
    });

})