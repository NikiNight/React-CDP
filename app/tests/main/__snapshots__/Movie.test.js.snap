// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test Movie component Movie matches snapshot 1`] = `
<article
  className="movie"
  onClick={[Function]}
>
  <div
    className="movie__image-wrapper"
  >
    <img
      alt=""
      className="movie__image"
    />
  </div>
  <div
    className="movie__description"
  >
    <div>
      <h2
        className="movie__title"
      />
      <p
        className="movie__category"
      />
    </div>
    <p
      className="movie__year"
    />
  </div>
</article>
`;
